fastqsplitter.wdl

```diff
@@ -0,0 +1,62 @@
+version 1.0
+
+# MIT License
+#
+# Copyright (c) 2019 Leiden University Medical Center
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+task Fastqsplitter {
+    input {
+        File inputFastq
+        Array[String]+ outputPaths
+        String dockerImage = "quay.io/biocontainers/fastqsplitter:1.0.0--py_0"
+        Int? compressionLevel
+        Int? threadsPerFile
+         # fastqplitter utilizes one thread per input file and one or more threads per output file + one thread for the application.
+         # Since a compression level of 1 is used, each output file uses approx 0.5 cores.
+        Int cores = 1 + ceil(0.5 * length(outputPaths))
+    }
+
+    command {
+        set -e
+        mkdir -p $(dirname ~{sep=') $(dirname ' outputPaths})
```

Comment:
Is it necessary to put `) $(dirname ` as separator? Wouldn't a simple space work?

Reply:
This is directly copied from biopet fastqsplitter. A simple space will not work as you do not have the needed parent directory structure.

Reply:
But you can use only one call to `dirname`, right? `dirname` takes multiple inputs:

```BASH
dcats@[...]$ dirname /some/absolute/path some/relative/path /blah/die/blah
/some/absolute
some/relative
/blah/die
```

Reply:
Did not know that. Will change!